hypothesis-python/src/hypothesis/internal/constants_ast.py

```diff
@@ -14,25 +14,56 @@
 import sys
 from ast import AST, Constant, Expr, NodeVisitor, UnaryOp, USub
 from functools import lru_cache
+from pathlib import Path
 from types import ModuleType
-from typing import TYPE_CHECKING, Optional, Union
+from typing import TYPE_CHECKING, Optional, TypedDict, Union
 
 from hypothesis.internal.escalation import is_hypothesis_file
-from hypothesis.internal.scrutineer import ModuleLocation
 
 if TYPE_CHECKING:
     from typing import TypeAlias
 
 ConstantT: "TypeAlias" = Union[int, float, bytes, str]
 
 
+class ConstantsT(TypedDict):
+    integer: set[int]
+    # we store floats in either their float or float_to_int form in different
+    # places.
+    float: Union[set[float], set[int]]
```

Comment:
I suggest that we _not_ do that.  

More specifically, I think we can ignore nans and -0.0 in the constants subsystem (other heuristics will upweight them), and thus stick to `float` here.

Reply:
yeah this is a lot cleaner